homeassistant/components/fritzbox/__init__.py

```diff
@@ -1,90 +1,71 @@
+from .const import CONF_CONNECTIONS, DOMAIN, SUPPORTED_DOMAINS
 
-DOMAIN = "fritzbox"
 
-ATTR_STATE_BATTERY_LOW = "battery_low"
-ATTR_STATE_DEVICE_LOCKED = "device_locked"
-ATTR_STATE_HOLIDAY_MODE = "holiday_mode"
-ATTR_STATE_LOCKED = "locked"
-ATTR_STATE_SUMMER_MODE = "summer_mode"
-ATTR_STATE_WINDOW_OPEN = "window_open"
+def ensure_unique_hosts(value):
+    """Validate that all configs have a unique host."""
+    vol.Schema(vol.Unique("duplicate host entries found"))(
+        [socket.gethostbyname(entry[CONF_HOST]) for entry in value]
+    )
+    return value
 
 
 CONFIG_SCHEMA = vol.Schema(
     {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_DEVICES): vol.All(
-                    cv.ensure_list,
-                    [
-                        vol.Schema(
-                            {
-                                vol.Required(CONF_HOST): cv.string,
-                                vol.Required(CONF_PASSWORD): cv.string,
-                                vol.Required(CONF_USERNAME): cv.string,
-                            }
-                        )
-                    ],
+        DOMAIN: vol.All(
+            cv.ensure_list,
+            [
+                vol.Schema(
+                    {
+                        vol.Required(CONF_HOST): cv.string,
+                        vol.Required(CONF_PASSWORD): cv.string,
+                        vol.Required(CONF_USERNAME): cv.string,
+                    }
                 )
-            }
+            ],
+            ensure_unique_hosts,
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up the fritzbox component."""
-
-    fritz_list = []
-
-    configured_devices = config[DOMAIN].get(CONF_DEVICES)
-    for device in configured_devices:
-        host = device.get(CONF_HOST)
-        username = device.get(CONF_USERNAME)
-        password = device.get(CONF_PASSWORD)
-        fritzbox = Fritzhome(host=host, user=username, password=password)
-        try:
-            fritzbox.login()
-            _LOGGER.info("Connected to device %s", device)
-        except LoginError:
-            _LOGGER.warning("Login to Fritz!Box %s as %s failed", host, username)
-            continue
-
-        fritz_list.append(fritzbox)
+async def async_setup(hass, config):
+    """Set up the AVM Fritz!Box integration."""
+    if DOMAIN in config:
+        for entry_config in config[DOMAIN]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": "import"}, data=entry_config
+                )
+            )
 
-    if not fritz_list:
-        _LOGGER.info("No fritzboxes configured")
-        return False
+    return True
 
-    hass.data[DOMAIN] = fritz_list
 
-    def logout_fritzboxes(event):
-        """Close all connections to the fritzboxes."""
-        for fritz in fritz_list:
-            fritz.logout()
```

Comment:
Should we still log out when home assistant stops?

Reply:
Yes, but I was unsure where I need this logout at the end. What is best: listen for `EVENT_HOMEASSISTANT_STOP` or implement `async_unload_entry`.

Reply:
Both are good to do.

Reply:
Should we add logout on home assistant stop event back?

Reply:
The `async_unload_entry` is not called on home assistant stop?

Reply:
No.